native/src/routes/language/containers/ChangeLanguageModalContainer.js

```diff
@@ -29,21 +30,47 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
   const currentLanguage: string = ownProps.navigation.getParam('currentLanguage')
   const languages: Array<LanguageModel> = ownProps.navigation.getParam('languages')
   const availableLanguages: Array<string> = ownProps.navigation.getParam('availableLanguages')
+  const previousKey = ownProps.navigation.getParam('previousKey')
+
+  const newsRouteMapping = state.cityContent?.newsRouteMapping
+  const newsType = previousKey && newsRouteMapping && newsRouteMapping[previousKey] && newsRouteMapping[previousKey].type
+
   return {
     currentLanguage,
     languages,
-    availableLanguages
+    availableLanguages,
+    newsType
   }
 }
 
-type DispatchType = Dispatch<SwitchContentLanguageActionType>
+type DispatchType = Dispatch<StoreActionType>
 const mapDispatchToProps = (dispatch: DispatchType, ownProps: OwnPropsType): DispatchPropsType => {
+  const cityCode = ownProps.navigation.getParam('cityCode')
+  const previousKey = ownProps.navigation.getParam('previousKey')
+
   return {
-    changeLanguage: (newLanguage: string) => {
+    changeLanguage: (newLanguage: string, newsType: ?NewsType) => {
       dispatch({
         type: 'SWITCH_CONTENT_LANGUAGE',
         params: { newLanguage, city: ownProps.navigation.getParam('cityCode'), t: ownProps.t }
       })
+
+      if (newsType) {
```

Comment:
So what happens if there is no news type?

Reply:
Nothing apart from `SWITCH_CONTENT_LANGUAGE` is executed which is enough for all other routes :)

Reply:
So there is no content displayed or are you routed to the CategoryList?

Reply:
For other routes? The data in the new language is fetched (see `watchSwitchContentLanguage`) and afterwards the routes are "morphed" to the new language (i.e. replacing the old models, params, ...) (see `morphContentLanguage`). During this time, a loading spinner is displayed. The route stays always the same, i.e. no redirecting e.g. to the CategoryList.

Not sure if I understood your question correctly :thinking:

Reply:
I feel it my question is also not directly dependend to this PR. I will ask you in Person about this next time.

Reply:
Okay :D